locale/fr_FR/LC_MESSAGES/django.po

```diff
@@ -2594,14 +2594,14 @@ msgstr "Livres"
 #: bookwyrm/templates/user/user_preview.html:13
 #, python-format
 msgid "Joined %(date)s"
-msgstr "Enregistré(e) %(date)s"
+msgstr "Membre depuis %(date)s"
```

Comment:
I’m usually browsing in English and just switched to French for testing…

Just raising the question that `%(date)s` will be replaced by `il y a [n] (jour(s)|mois|année(s)]`, so _Membre depuis_ is going to be a bit weird (to me) although correct (I think). I don’t know what kind of tweaking is possible for the formatting of `date` in that case but maybe @mouse-reeve knows.

Reply:
> Just raising the question that `%(date)s` will be replaced by `il y a [n] (jour(s)|mois|année(s)]`

Ah yes, in my mind I thought I had removed `il y a`, so _« Membre depuis 2 jours »_ would have worked. I don't know if we'll be able to use a different translation for the `|naturaltime` filter

Reply:
Yep… I thought the same when translating. :)

I do like _Membre_ better though, I wonder if @mouse-reeve would have anything against updating the logic and use something less verbose like `Member since: {2021-04-17}` or `Member for {3 months}`, for example.

I would guess the logic is to easily read for how long the account has been a member of the instance; some sort of trust factor that is easier to read than computing a date difference. Then again… Just an assumption.

Reply:
In French we'd like to use `date|timesince` instead of `date|naturaltime`. Do you know if it's possible to give the choice of formats to the translator?

Reply:
I don't think it's possible to change it just for one locale, but I have no objection whatsoever to using `timesince` across the board here

Reply:
I updated the file and the French translation. Will that cause a problem for other translators?

Reply:
hm, I realized I actually misread the initial problem here with `naturaltime` - sorry, I've been kinda scatterbrained the for the last week. It would cause problems for other locales to switch to `timesince`, but would `naturalday` work instead? There's no need as far as I can see to be precise about how many hours ago someone joined, I think `naturalday` doesn't add the "ago"/"il y a", and seems to be compatible with the existing translations?

Reply:
Here's the current situation

![image](https://user-images.githubusercontent.com/1781070/115590287-1179ed80-a2d1-11eb-85fa-9805db3f313b.png)

`naturaltime` is what's used right now, and my problem was that it adds "ago"/"il y a", so my proposal to change _« Enregistré(e) il y a 3 jours »_ (= "Joined 3 days ago") to _« Membre depuis 3 jours »_ (= "Member for 3 days") won't work, it'll give us _« Membre depuis il y a trois jours »_ (= "Member for 3 days ago").

Do you think that we can specify how the `naturaltime` filter works for French? I don’t think that filters can be applied inside translation files.

Reply:
(I proposed to change the current translation because using a verb in French requires to adapt it to the gender of the user, or use a possibly inaccessible solution with an _(e)_ ending. "Member for 3 days" is the best neutral way to convey the meaning)

Reply:
right, I think `naturalday`, as opposed to `naturaltime`, will _not_ add `ago"/"il y a"`. 

With `naturaltime`: "Joined today"
With `naturalday`: "Joined an hour ago"

With `naturaltime`: "Joined 2 weeks, 1 day ago"
With `naturalday`: "Joined April 6, 2021"

I think that solves the issues with "ago" in French without changing other locales (it at least doesn't require any wording changes in English). I hope I'm understanding this correctly?

Reply:
I hadn't read `naturalday`, I thought it was `naturaltime`, sorry about that 🤦

Reply:
After testing a bit and reading the source code for `naturalday`, it won't work well with French. Before dates you need the article _« le »_, but before relative times (_« hier »_, "yesterday") you don't need it. `naturalday` alternates between the two, without adding the article, so it won't work.

What could work well in French would be `naturaldelta` from Python Humanize:

> This is similar to `naturaltime`, but does not add tense to the result.

https://python-humanize.readthedocs.io/en/latest/time/#humanize.time.naturaldelta

Sadly it's unavailable, and I think it would cause the same problems as using directly `timesince`.

Back to square one, I guess :)

Reply:
dang! too bad, thank you for looking into it.

Reply:
> […] It would cause problems for other locales to switch to timesince, […]

@mouse-reeve: Do you have specific problems in mind?

I’m having in mind a less verbose form like `Member for: {2021-04-22}`, for example. There would be less possible variations and the string would focus more on translating the text than the generated form of the date. That way, I understand `naturalday` would work as intended.

Does it seem too easy of a solution that I missed something or is someone against lists? :)

Reply:
Ping… :)